crates/database/src/solver_competition.rs

```diff
@@ -1,29 +1,30 @@
-use crate::TransactionHash;
+use crate::{auction::AuctionId, TransactionHash};
 use sqlx::{types::JsonValue, PgConnection};
 
-pub type SolverCompetitionId = i64;
-
 pub async fn save(
     ex: &mut PgConnection,
+    id: AuctionId,
     data: &JsonValue,
     tx_hash: Option<&TransactionHash>,
-) -> Result<SolverCompetitionId, sqlx::Error> {
+) -> Result<(), sqlx::Error> {
     const QUERY: &str = r#"
-INSERT INTO solver_competitions (json, tx_hash)
-VALUES ($1, $2)
-RETURNING id
-    "#;
-    let (id,) = sqlx::query_as(QUERY)
+INSERT INTO solver_competitions (id, json, tx_hash)
+VALUES ($1, $2, $3)
+ON CONFLICT (id) DO UPDATE
+SET json = EXCLUDED.json, tx_hash = EXCLUDEd.tx_hash
```

Comment:
This gives us the possibility of overwriting the competition info. Could also leave it out so we get an error if we try to. Maybe error is better because we're not currently intentionally overwriting?

Reply:
> Maybe error is better because we're not currently intentionally overwriting?

This would be my inclination. Especially because:

> In practice this will not happen because all the auction ids are already larger than the competition ids.

So if it does, then one of our assumptions is broken.